_posts/2016-01-20-set-up-and-run-hubot-in-non-heroku-server.markdown

```diff
@@ -0,0 +1,102 @@
+---
+layout: post
+title: "Set up and run Hubot without using Heroku"
+date: 2016-01-20 11:16:00
+categories: ["hubot", "linux"]
+author: "mauro-oto"
+---
+
+[Hubot](https://hubot.github.com) makes it incredibly easy to setup on a Heroku
+server, by taking advantage of its Procfile. Simply running
+`git push heroku master` deploys the app and starts it.
+
+When it comes to deploying to your own Linux server, given that
+`foreman` doesn't really like background processes (see:
+[ddollar/foreman#65](https://github.com/ddollar/foreman/issues/65)), you need to
+use something like `monit`, `systemd` or `tmux` to better manage your Hubot
+process.
+
+<!--more-->
+
+If you don't already have your own Hubot repository, you can easily create one.
+The easiest way is using the hubot generator:
+
+```bash
+npm install -g yo generator-hubot
+```
+
+Run the generator in a folder of your choosing:
+
+```bash
+$ mkdir myhubot
+$ cd myhubot
+$ yo hubot
+```
+
+Go through the steps the generator takes you, and when you're done, initialize a
+Git repository and push it to Github or your preferred Git server.
+
+Go ahead and choose where you want to run your Hubot from and SSH into your
+instance (we used an AWS EC2 instance). Once in it, you can clone your Hubot
+repository, or if you prefer, create a Capistrano recipe and deploy it.  
+
+As I mentioned before, you could run `foreman start` now to check that your bot
+is working, but you can't leave it running in the background out-of-the-box, as
+`foreman` is not friendly to background processes.
+
+You can work around this by leaving `foreman` itself running in a dettached tmux
+session:
+
+```bash
+$ tmux new -s hubot
+$ cd /path/to/hubot
+$ foreman start
+<Ctrl-b>-d
+```
+
+This is **not the best way to do it**, as the process could end unexpectedly and
+it won't come up again, making you re-attach to the session or kill it and
```

Comment:
@mauro-oto I'd split this in two sentences. 

"This is **not the best way to do it**, as the process could end unexpectedly and it wouldn't come up again."

"This would force you to re-attach to the session to restart Hubot manually."